CMakeLists.txt

```diff
@@ -15,8 +15,12 @@ if (CMAKE_BUILD_TYPE STREQUAL "Release")
     set(FLAGS "-DNDEBUG ${OPT}")
 elseif (CMAKE_BUILD_TYPE STREQUAL "Debug")
     set(FLAGS "-g -O0")
-else ()
+elseif (CMAKE_BUILD_TYPE STREQUAL "VTune")
+    set(FLAGS "-g -O2 -shared-libgcc")
+elseif (CMAKE_BUILD_TYPE STREQUAL "Assert")
```

Comment:
maybe we can reset `CMAKE_C_FLAGS_RELEASE ` and `CMAKE_CXX_FLAGS_RELEASE`  to enable assert in release mode